### ISSUE TITLE
Something something p5.js

### ISSUE THREAD

**Issue by jonocarroll** [ropensci/ozunconf17#23]
I don't have a specific idea in mind, but [`p5.js`](https://p5js.org/examples/) is a fantastic library that just begs to be used in a `htmlwidget`. Some 'prior art': 

geom_realtime(): live-plotting microphone input: https://www.youtube.com/watch?v=3mdiCUbgxi0

click and animate: https://www.youtube.com/watch?v=BRYlPWEtVQ8

The examples page linked above has many, many possibilities.

**Comment by PeteHaitch** [comment]
Sean Kross (@seankross) gave a really cool talk at JSM on https://github.com/seankross/p5, his R package with bindings to p5. His slides are available as a blog post http://seankross.com/2017/08/11/Beyond-Axes-Simulating-Systems-with-Interactive-Graphics.html

**Comment by jonocarroll** [comment]
So, yeah. I want to make what Sean made, I guess.

It doesn't look like he took it anywhere near as far as it could go, so there's still plenty more that could build on this. Take that repo as inspiration for R bindings to p5.js and think of what we could do with it.

**Comment by PeteHaitch** [comment]
Sean may be interested in collaborating to advance the package. He's just started grad school, so I'm sure he's got a billion other things on his plate!

**Comment by adam-gruer** [comment]
It could be awesome to sit this over the data coming out of #3

**Comment by seankross** [comment]
Hey all, (thanks for the mention @PeteHaitch) 

I am just starting grad school and my life is a little crazy as a result! Still I really think that p5 and R should absolutely get together. @jonocarroll if you want to fork my repository and run with the code then go for it! If you want to start from scratch because my library doesn't make any sense you should do that, don't worry about stepping on my toes I don't care! If you want to collaborate on what I've already written I'd be happy to do that as well, though I'm not going to be able to be online during ozunconf17.

**tl;dr** Go for it and don't worry about me but I'd be happy to help too!

Sean

**Comment by MilesMcBain** [comment]
Ohhhhhh wow. This is great stuff... also I really did that slideshow/blog post mashup thing you have going on @seankross. I feel like I watched your talk now.

**Comment by seankross** [comment]
Yeah another "someday maybe" project is to turn that slide hangout presentation format into a proper R Markdown template using the same HTML-comment-as-content framework as [Ari](https://github.com/seankross/ari).

**Comment by jonocarroll** [comment]
Cheers @seankross -- I'm absolutely keen to collaborate on this stuff when you get some bandwidth. Your `p5` functions look like and do exactly what I had in mind initially and I'm definitely going to base what I do on your work. My larger goal though is to enable a user to generate plots with the same R syntax, as in #26

**Comment by jonocarroll** [comment]
Right, so with `geom_realtime()` as its own project/issue (#26) this one needs more focus. I say we extend what @seankross has so brilliantly made (https://github.com/seankross/p5) into a more feature-rich wrapper. 

I don't have a specific use case yet, but it would be nice to wrap some of the features of `p5`. As a lofty goal, I would think we could get a game of FlappyBird made during the unconf using only R commands and R wrappers to `p5`. [Here's a pure JS implementation](https://github.com/loveky/FlappyBird-in-pure-JavaScript/blob/master/js/game.js) that we could map the functionality of.

A slightly less silly/more useful goal would be to do the wrapping so that a user could plot *their* data as shapes/some sort of interactivity/3D.

**Comment by hrbrmstr** [comment]
Just making sure folks are also aware of https://processing-r.github.io/

**Comment by jonocarroll** [comment]
I was not aware of it, so thank you @hrbrmstr! It's definitely a more sophisticated application of the bindings, but the `p5` package did exactly what we wanted and allowed us to extend to arbitrary `R` function input processing data streams, plotting in real-enough-time.

Vignettes and blog posts to follow in good time.

**Comment by stefaniebutland** [comment]
Putting @jonocarroll's blog post on `realtime` here since references things discussed here: 2017 rOpenSci ozunconf :: Reflections and the realtime Package https://ropensci.org/blog/2017/11/14/realtime/

**Comment by bedantaguru** [comment]
@jonocarroll so what is the plan for `geom_realtime`
